CoreFunctionality/test/unit/src/au/gov/asd/tac/constellation/functionality/ConstellationErrorManagerNGTest.java

```diff
@@ -0,0 +1,139 @@
+ * @author OrionsGuardian
+ */
+public class ConstellationErrorManagerNGTest {
+    private static final Logger LOGGER = Logger.getLogger(ConstellationErrorManagerNGTest.class.getName());
+    
+    @Test
+    public void runExceptionsTest() {
+        final MockedStatic<ErrorReportDialogManager> erdmStatic = Mockito.mockStatic(ErrorReportDialogManager.class);
+        final ErrorReportDialogManager erdm = mock(ErrorReportDialogManager.class);
+        erdmStatic.when(ErrorReportDialogManager::getInstance).thenReturn(erdm);
+
+        final ConstellationErrorManager errorManager = new ConstellationErrorManager();
+        LOGGER.addHandler(errorManager);
+        LOGGER.setLevel(Level.ALL);
+        
+        // generate exception log messages
+        simulateException(Level.SEVERE);
+        simulateException(Level.WARNING);
+        simulateException(Level.INFO);
+        simulateException(Level.FINE);
+
+        LOGGER.log(Level.INFO, "\n ------- start wait for data ... {0}", new Date());
+        boolean dataAvailable = waitForDataToBeAvailable(Level.SEVERE, 1);
+        assertTrue(dataAvailable);
+        LOGGER.log(Level.INFO, "\n ------- finished waiting for data ... {0}", new Date());
+        
+        // confirm the exceptions have been stored in the SessionData class
+        confirmExceptionStored(Level.SEVERE, 1);
+        confirmExceptionStored(Level.WARNING, 1);
+        confirmExceptionStored(Level.INFO, 1);
+        confirmExceptionStored(Level.FINE, 1);
+        
+        confirmEntryOccurrences(Level.SEVERE, 1);
+        
+        // NOTE: the following 2 calls NEED to be on the SAME LINE to generate identical stack traces
+        simulateException(Level.SEVERE); simulateException(Level.SEVERE);
+        
+        LOGGER.log(Level.INFO, "\n ------- 2.start wait for data ... {0}", new Date());
+        dataAvailable = waitForDataToBeAvailable(Level.SEVERE, 2);
+        assertTrue(dataAvailable);
+        LOGGER.log(Level.INFO, "\n ------- 2.finished waiting for data ... {0}", new Date());
+
+        // confirm correct number of entries and occurrences
+        confirmExceptionStored(Level.SEVERE, 2);
+        confirmEntryOccurrences(Level.SEVERE, 2);
+    }
+    
+    public void simulateException(final Level logLevel){
+        LOGGER.log(Level.INFO, "\n ------- simulating {0} exception", logLevel.getName());
+        final Exception e = new Exception("Something totally not unexpected happened !");
+        LOGGER.log(logLevel, "Simulating a " + logLevel.getName() + " exception !", e);
+        LOGGER.info("\n ------- simulated.");
+    }
+    
+    public void delay(final int milliseconds){
+        // may need to wait for the error handler to do it's thing
+        try {
+            Thread.sleep(milliseconds); //NOSONAR
+        } catch (final InterruptedException ex) {
+            LOGGER.log(Level.INFO, "\n -------- sleep disturbed: {0}", ex.toString());
+        }
+    }
```

Comment:
can we look to use a CompletableFuture instead? No way to tell whether an arbitrary delay is long enough (but not too long) in any given system environment